src/ImageSharp/Formats/Png/Filters/AverageFilter.cs

```diff
@@ -22,18 +22,23 @@ internal static unsafe class AverageFilter
         /// <param name="bytesPerScanline">The number of bytes per scanline</param>
         /// <param name="bytesPerPixel">The bytes per pixel.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void Decode(byte[] scanline, byte[] previousScanline, int bytesPerScanline, int bytesPerPixel)
+        public static void Decode(ref byte scanline, ref byte previousScanline, int bytesPerScanline, int bytesPerPixel)
```

Comment:
Not sure if I understand this right now, I'll have another look tomorrow. If possible, I prefer span on the API-s.